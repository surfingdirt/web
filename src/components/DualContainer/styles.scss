@import "../../styles/colors";
@import "../../styles/boxmodel";

.container {
  display: flex;
  flex-wrap: wrap;
  align-items: baseline;
  justify-content: space-between;
}

/* Extends */

%rightElementContainer {
  display: flex;
  align-items: center;
  .title {
    margin-right: $margin-small-2;
    font-weight: bold;
  }
  svg {
    fill: currentColor;
  }
}
