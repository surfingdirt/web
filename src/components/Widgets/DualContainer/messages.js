const messages = (getText, getPlural) => (key) =>
  ({
    followUs: getText(
      'Follow Us',
      'DualContainer',
    ) /* Title of a section containing a list of social media links */,
    viewAll: getText('View all', 'DualContainer') /* Title of a button to expand a list */,
  }[key]);

export default messages;
