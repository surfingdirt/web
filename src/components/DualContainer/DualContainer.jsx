import PropTypes from 'prop-types';
import React from 'react';

import styles from './styles.scss';

const DualContainer = ({ children }) => <div className={styles.container}>{children}</div>;

DualContainer.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
};

export default DualContainer;
