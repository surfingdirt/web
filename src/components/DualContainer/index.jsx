import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import styles from './styles.scss';

const DualContainer = ({ children, nowrap }) => (
  <div className={classnames(styles.container, { [styles.nowrap]: nowrap })}>{children}</div>
);

DualContainer.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
  nowrap: PropTypes.bool,
};

DualContainer.defaultProps = {
  nowrap: false,
};

export default DualContainer;
