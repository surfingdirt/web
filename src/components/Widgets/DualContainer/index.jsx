import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import styles from './styles.scss';

const DualContainer = ({ children, className, nowrap }) => (
  <div className={classnames(styles.container, { [styles.nowrap]: nowrap }, className)}>
    {children}
  </div>
);

DualContainer.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
  className: PropTypes.string,
  nowrap: PropTypes.bool,
};

DualContainer.defaultProps = {
  className: null,
  nowrap: false,
};

export default DualContainer;
