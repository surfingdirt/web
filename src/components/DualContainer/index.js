import DualContainer from './DualContainer';

export default DualContainer;
