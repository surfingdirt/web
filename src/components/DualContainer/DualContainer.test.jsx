import React from 'react';

import DualContainer from './DualContainer';

describe('Dual Container', () => {
  it('should render a dual container with only left element', () => {
    const wrapper = mount(
      <DualContainer>
        <span>Left Element</span>
      </DualContainer>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a dual container with left & right elements', () => {
    const wrapper = mount(
      <DualContainer>
        <span>Left Element</span>
        <span>Right Element</span>
      </DualContainer>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
