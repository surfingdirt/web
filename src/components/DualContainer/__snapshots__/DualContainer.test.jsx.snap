// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dual Container should render a dual container with left & right elements 1`] = `
<div
  className="container"
>
  <span>
    Left Element
  </span>
  <span>
    Right Element
  </span>
</div>
`;

exports[`Dual Container should render a dual container with only left element 1`] = `
<div
  className="container"
>
  <span>
    Left Element
  </span>
</div>
`;
